We should also change the phpdoc of `DataObject\Service::getValidLayouts()` to this:
```
    /**
     * @return array<string, DataObject\ClassDefinition\CustomLayout>
     */
    public static function getValidLayouts(Concrete $object): array
    {
```
With this change phpstan would noticed this error:
```
 ------ ------------------------------------------------------------------------------------------------------------
  Line   bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php
 ------ ------------------------------------------------------------------------------------------------------------
  513    Offset 0 does not exist on non-empty-array<string, Pimcore\Model\DataObject\ClassDefinition\CustomLayout>.
 ------ ------------------------------------------------------------------------------------------------------------

 [ERROR] Found 1 error
```

--- a/models/DataObject/Service.php
+++ b/models/DataObject/Service.php
@@ -986,7 +986,7 @@ class Service extends Model\Element\Service
     /**
      * @param Concrete $object
      *
-     * @return DataObject\ClassDefinition\CustomLayout[]
+     * @return array<string, DataObject\ClassDefinition\CustomLayout>
      */
     public static function getValidLayouts(Concrete $object)
     {

